dataprofiler/profilers/data_labeler_column_profile.py

```diff
@@ -307,6 +307,48 @@ def profile(self) -> dict:
         }
         return profile
 
+    @classmethod
+    def load_from_dict(cls, data, options: dict | None = None) -> BaseColumnProfiler:
+        """
+        Parse attribute from json dictionary into self.
+
+        :param data: dictionary with attributes and values.
+        :type data: dict[string, Any]
+        :param options: options for loading column profiler params from dictionary
+        :type options: Dict | None
+
+        :return: Profiler with attributes populated.
+        :rtype: DataLabelerColumn
+        """
+        opt = DataLabelerOptions()
+        data_labeler_load_attr = data.pop("data_labeler")
+        if "from_library" in data_labeler_load_attr:
+            opt.data_labeler_object = DataLabeler.load_from_library(
+                data_labeler_load_attr["from_library"]
+            )
+        elif "from_disk" in data_labeler_load_attr:
+            raise NotImplementedError(
+                "Models intialized from disk have not yet been made deserializable"
+            )
+        else:
+            raise ValueError(
+                "Deserialization cannot be done on labelers without "
+                "_default_model_loc set to known value."
+            )
+
+        # This is an ambiguous call to super classes.
+        # If load_from_dict is part of both super classes there may be issues
+        profile = super().load_from_dict(data, options=opt)
```

Comment:
I suggest updating this to be:
```python
profile = super().load_from_dict(data, options={cls.__name__: opt})
```

Reply:
In base we can do something like:
```python
if options is None:
            options = {}
        
class_options = options.get(cls.__name__)
profile = cls(data["name"], class_options)
```

Reply:
The above for me solved all the tests.
However, mypy decided to get a little talkative. 
Unfortunately, a proper solve requires generics. Something that was merged into main, but not here.
Which means, we likely need to partially update for generics from #840 in this and will have to resolve rebase on to main conflicts when they arise in the future.

Reply:
My suggestions to solve (don't include the random comments): 

`base_column_profilers.py`
```python
# in imports we need:
# add the extra imports to typing
from typing import Any, Callable, Generic, TypeVar

# refactor this import to be relative and add the BaseOption
from .profiler_options import BaseInspectorOptions, BaseOption

# add the Generic for any BaseColumn
BaseColumnProfilerT = TypeVar("BaseColumnProfilerT", bound="BaseColumnProfiler")

# update the class  to use generics, remove the typing ignore :)
class BaseColumnProfiler(Generic[BaseColumnProfilerT], metaclass=abc.ABCMeta):
    
    # update init, takes a BaseOption now... likely a more appropriate way to do this, but for now works
    def __init__(self, name: str | None, options: BaseOption | None = None):
        ...

    .
    .
    .
    # fix load from dict
    @classmethod
    def load_from_dict(
        cls: type[BaseColumnProfilerT],
        data: dict[str, Any],
        options: dict | None = None,
    ) -> BaseColumnProfilerT:
        """
        Parse attribute from json dictionary into self.

        :param data: dictionary with attributes and values.
        :type data: dict[string, Any]

        :return: Profiler with attributes populated.
        :rtype: BaseColumnProfiler
        """
        if options is None:
            options = {}

        class_options = options.get(cls.__name__)
        profile: BaseColumnProfilerT = cls(data["name"], class_options)
        .
        .
        .
```



`data_labeler_column_profile.py`
```python
# update the class
class DataLabelerColumn(BaseColumnProfiler["DataLabelerColumn"]):
    .
    .
    .
    # fi load_from_dict to specify proper output class
    @classmethod
    def load_from_dict(cls, data, options: dict | None = None) -> DataLabelerColumn:
         # other fixes in here as mentioned above for getting the profile
     .
     .
     .
```

`json_decoder.py`
```python
# new imports to add
from typing import TYPE_CHECKING, Dict, Type


def load_column_profile(serialized_json: dict) -> "BaseColumnProfiler":
    .
    .
    .
    # updating the expected type here, i think this is still incorrect, but unsure proper way to generic it; fine for now
    column_profiler_cls: Type[
        "BaseColumnProfiler[BaseColumnProfiler]"
    ] = get_column_profiler_class(serialized_json["class"])
    return column_profiler_cls.load_from_dict(serialized_json["data"])
.
.
.
```

Reply:
in `json_decoder.py`,
can also fix cases of `Optional[...]` ->  `... | None`

Reply:
Will be fun when we have to do these same fixes for the compiler maybe...